File: program/program/find-the-transpose-of-a-matrix/FindTheTransposeOfAMatrix.java
class FindTheTransposeOfAMatrix {
    public static void main(String[] args) {
        int matrix[][] = { { 1, 2, 3 }, { 4, 5, 6 }, { 7, 8, 9 } };
        int size = matrix.length;

        for (int i = 0; i < size; i++)
            for (int j = i + 1; j < size; j++) {
                int help = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = help;
            }

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++)
                System.out.print(matrix[i][j] + " ");
            System.out.println("");
        }
    }
}

File: program/program/find-the-transpose-of-a-matrix/find_the_transpose_of_a_matrix.go
package main

import "fmt"

func transpose(matrix [][]int) [][]int {
	rows := len(matrix)
	cols := len(matrix[0])

	transposed := make([][]int, cols)
	for i := 0; i < cols; i++ {
		transposed[i] = make([]int, rows)
	}

	for i := 0; i < rows; i++ {
		for j := 0; j < cols; j++ {
			transposed[j][i] = matrix[i][j]
		}
	}

	return transposed
}

func main() {
	matrix := [][]int{
		{1, 2, 3},
		{4, 5, 6},
		{7, 8, 9},
	}

	transposed := transpose(matrix)

	fmt.Println("Original Matrix:")
	printMatrix(matrix)

	fmt.Println("\nTransposed Matrix:")
	printMatrix(transposed)
}

func printMatrix(matrix [][]int) {
	for _, row := range matrix {
		for _, val := range row {
			fmt.Printf("%d\t", val)
		}
		fmt.Println()
	}
}
